src/Type/PostObject/Mutation/PostObjectCreate.php

```diff
@@ -122,6 +147,43 @@ public static function mutate( \WP_Post_Type $post_type_object ) {
 					 */
 					PostObjectMutation::update_additional_post_object_data( $post_id, $input, $post_type_object, $mutation_name );
 
+					/**
+					 * Determine whether the intended status should be set or not.
+					 *
+					 * By filtering to false, the $intended_post_status will not be set at the completion of the mutation.
+					 *
+					 * This allows for side-effect actions to set the status later. For example, if a post
+					 * was being created via a GraphQL Mutation, the post had additional required assets, such as images
+					 * that needed to be sideloaded or some other semi-time-consuming side effect, those actions could
+					 * be deferred (cron or whatever), and when those actions complete they could come back and set
+					 * the $intended_status.
+					 *
+					 * @param boolean $should_set_intended_status Whether to set the intended post_status or not. Default true.
+					 * @param \WP_Post_Type $post_type_object The Post Type Object for the post being mutated
+					 * @param string $mutation_name The name of the mutation currently in progress
+					 * @param string $intended_post_status The intended post_status the post should have according to the mutation input
+					 * @param string $default_post_status The default status posts should use if an intended status wasn't set
+					 */
+					$should_set_intended_status = apply_filters( 'graphql_post_object_create_should_set_intended_post_status', true, $post_type_object, $mutation_name, $intended_post_status, $default_post_status  );
+
+					/**
+					 * If the intended post status and the default post status are not the same,
+					 * update the post with the intended status now that side effects are complete.
+					 */
+					if ( $intended_post_status !== $default_post_status && true === $should_set_intended_status ) {
+
+						$update_args = [
+							'ID' => $post_id,
+							'post_status' => $intended_post_status,
+							// Prevent the post_date from being reset if the date was included in the create post $args
+							// see: https://core.trac.wordpress.org/browser/tags/4.9/src/wp-includes/post.php#L3637
+							'edit_date' => ! empty( $post_args['post_date'] ) ? $post_args['post_date'] : false,
+						];
+
+						$update_args = array_merge( $update_args );
+						wp_update_post( $update_args );
```

Comment:
Shouldn't this get moved out of this if statement so it fires even if the filter is set to false?

Reply:
No. The intent is that a plugin, like DFM GraphQL Extensions, that needs to process sideloaded images, can filter this to false, meaning the created post will remain a draft. Then the process that sideloads the images can come back and set the post as `Publish` or whatever the intended status was. 

Otherwise, if the status was set to publish and the deferred (cron / queued tasks, etc) side effects were not complete, then you'd have a published post with incomplete data.